### PR TITLE
Apply defaults to config inside adapter, without the config provider involved

### PR DESCRIPTION
**Pull request by roblourens**
It won't be invoked when old node-debug dispatches to pwa-node, or for the VS case
- And always activate when node-debug activates
- And add "request" back for tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,6 +99,7 @@
     "npm-run-all": "^4.1.5",
     "nyc": "^14.1.1",
     "puppeteer": "^1.20.0",
+    "request": "^2.88.0",
```
Comment (connor4312) [inline package.json]: Was this intentional?

Comment (roblourens) [reply package.json]: Yes, it's needed for tests (peerDependency of "request-promise-native")

**Comment by roblourens** [comment]
Actually, should I remove the code from the debug config providers that does the same thing?

**Comment by connor4312** [comment]
The default configs have some portions that reference the workspace folder that vs code needs to resolve. I think it's fine as a stepping stone here but it might not always be safe to fill in the defaults outside of the config provider.

**Comment by roblourens** [comment]
I realized that after some debugging, we won't resolve `${workspaceFolder}` here. I think I should still merge this for the VS scenario but they should know if they need to pass `rootPath`. Also I think we should merge rootPath/outFiles.

**Comment by connor4312** [comment]
Sounds good.

In addition to the breakpoint predictor, `rootPath` is used as a fallback for some other scenarios but I don't _think_ it's ever needed/hit there with our fulfilled defaults. I agree we should probably coalesce it into something more specific such as outFiles.

**Comment by roblourens** [comment]
I'm not sure what you're referring to

**Comment by connor4312** [comment]
`assignWorkspaceFolder` in `attachProcess` is a special-case equivalent of `resolveVariablesInConfig`. I guess not entirely remove the fact that we need it, but deduplicate it at least.

**Pull request merged**